odoh.go

```diff
@@ -215,25 +218,30 @@ func (k ObliviousDoHConfigContents) CipherSuite() (hpke.CipherSuite, error) {
 	return hpke.AssembleCipherSuite(k.KemID, k.KdfID, k.AeadID)
 }
 
-type ObliviousDNSKeyPair struct {
+type ObliviousDoHKeyPair struct {
 	Config    ObliviousDoHConfig
-	SecretKey hpke.KEMPrivateKey
-	Seed      []byte
+	secretKey hpke.KEMPrivateKey
+	seed      []byte
 }
 
-func (k ObliviousDNSKeyPair) CipherSuite() (hpke.CipherSuite, error) {
-	return hpke.AssembleCipherSuite(k.Config.Contents.KemID, k.Config.Contents.KdfID, k.Config.Contents.AeadID)
+func (k ObliviousDoHKeyPair) SerializePrivateKey() ([]byte, error) {
```

Comment:
Serialization/Deserialization is error prone, we should have a test for being able to recreate the PrivateKey during a transaction.